compiler/rustc_mir_transform/src/pass_manager.rs

```diff
@@ -198,6 +198,22 @@ fn run_passes_inner<'tcx>(
     let overridden_passes = &tcx.sess.opts.unstable_opts.mir_enable_passes;
     trace!(?overridden_passes);
 
+    let named_passes: FxIndexSet<_> =
+        overridden_passes.iter().map(|(name, _)| name.as_str()).collect();
+
+    for &name in named_passes.difference(&*crate::PASS_NAMES) {
+        tcx.dcx().emit_warn(errors::UnknownPassName { name });
+    }
```

Comment:
This *could* all be behind debug_assertions but since the common case is that `overriden_passes` is an empty list, this should have no effect

Reply:
No this is fine.